### PR TITLE
Fix bad connectivity

### LINKED ISSUE #61: Could not connect to latest block
When minimizing Metoronome it creates a pool of 'could not connect to latest block'. Cleans itself up quickly but the the in-app notifications should not accumulate so much. Version 0.3.0

### LINKED ISSUE #86: Detach listeners on main window reload

### LINKED ISSUE #123: Reset subscription if no blocks are seen in 1'
If the connection is interrupted, websocket provider does not always reconnect. To detect that case, should watch for incoming blocks. If no block in the last minutes, should restart the subscription.

### LINKED ISSUE #125: Connection error
From Matt L: 
I got the “Unhandled Promise Rejection” error when I minimized the wallet
When I brought the wallet back up, I had to login again (disconnected)

### PR DESCRIPTION
**Pull request by gabmontes**
And many other fixes here and there.

Hopefully closes #61, closes #86, closes #123 and , closes #125.

### CODE REVIEW

File: public/main/lib/time-bombs/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      debug('(%d) Boom!', bombId)
+
+      if (boomHandler) {
+        boomHandler()
+      } else {
+        throw new Error('Time bomb exploded!')
+      }
+    }, ms)
+
+    debug('(%d) Timer set to %dms', bombId, ms)
```
Comment (ianaya89) [inline public/main/lib/time-bombs/index.js:38]: Should we use `logger` this instead of `debug`?

**Review by ianaya89 — APPROVED** [review]
🥇

File: public/main/lib/time-bombs/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    return ticker
+  }
+
+  function onBoom (handler) {
```
Comment (pablen) [inline public/main/lib/time-bombs/index.js:43]: 😂

**Review by pablen — APPROVED** [review]

File: public/main/lib/time-bombs/index.js (hunk 1 above)
Comment (gabmontes) [reply public/main/lib/time-bombs/index.js:38]: I don't think so. The plan is to make a lib out of it, so no logger.

File: public/main/lib/time-bombs/index.js (hunk 2 above)
Comment (gabmontes) [reply public/main/lib/time-bombs/index.js:43]: 😆

**Pull request merged**